src/Features/LanguageServer/ProtocolUnitTests/CodeLens/CSharpCodeLensTests.cs

```diff
@@ -231,4 +231,56 @@ void UseM()
         var firstDocumentResult2 = await testLspServer.ExecuteRequestAsync<LSP.CodeLens, LSP.CodeLens>(LSP.Methods.CodeLensResolveName, firstCodeLens, CancellationToken.None);
         Assert.NotNull(firstDocumentResult2?.Command?.Title);
     }
+
+    [Theory, CombinatorialData]
+    public async Task TestHasRunTestsCommandAsync(bool mutatingLspWorkspace)
```

Comment:
Could you add another two tests covering the basic scenario of NUnit and MSTest?

Reply:
I think for now I'll skip these:
1.  The nunit / mstest finding is already exercised in a bunch of other tests at a lower layer, and the codelens code itself doesn't know about them (the command is entirely generic)
2.  I want to avoid another round of CI here.

I'll look at adding them when I do the debugging part.